core/src/main/java/org/apache/calcite/sql/SqlCreate.java

```diff
@@ -26,30 +26,30 @@
 public abstract class SqlCreate extends SqlDdl {
 
   /** Whether "OR REPLACE" was specified. */
-  boolean replace;
+  final SqlCreateOrReplace replace;
```

Comment:
This field name is a little misleading now that it has a type that can be one of three options. We should rename this field to "createType" or "createOption" (depending on your preference regarding the name of the enum).

Reply:
I considered both of those, but they would also be confusing because SqlCreateType and SqlCreateOption both show up elsewhere in the code (presumably corresponding to CREATE TYPE and CREATE OPTION). I also don't love the name SqlCreateOrReplace, definitely open to other suggestions.

Reply:
How does "SqlCreateReplace" sound? Also, I would suggest renaming the variable holding this enum value too, since it is no longer a boolean and has either true or false. For the variable, I'll suggest the name of "option" with some prefixes? Such variable name might not be accurate within the project context, but would be very clear within the scope of the class.

Reply:
If CreateType and CreateOption are already taken, how about one of the following for the name of the enum?

- CreateVariant
- CreateModifier
- CreateSpecifier

Reply:
CreateSpecifier sounds good.